h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -438,7 +439,13 @@ void sendBuffer(){
   
     // Open channel on first write attempt
     private ByteChannel openChan() throws IOException {
-      return H2ONode.openChan(TCPReceiverThread.TCP_SMALL,_socketFactory, _key.getAddress(), _key.getPort());
+      try {
+        return H2ONode.openChan(TCPReceiverThread.TCP_SMALL, _socketFactory, _key.getAddress(), _key.getPort());
+      }catch (ConnectException e){
+        // can't establish connection to this node, report this as closed connection ( i.e the target h2o node is down)
+        _connection_closed = true;
+        throw e;
```

Comment:
He is trying to distinguish between two situations - network timeout and client closed comm. channel (that's interesting situation for _watch dog client_ since it should result into cluster shutdown). But agree, the state should be described in more explicit way.